File: frontend/hr-client/src/app/pojo/personal-info/EmergencyContact/EmergencyContact.ts
import { Address } from "../address/address";

export class EmergencyContact {

    fullName : String;
    phone: String;
    address : Address;

    constructor(
        fullName : String,
        phone: String,
        address : Address
    ){
        this.fullName = fullName;
        this.phone = phone;
        this.address = address;
    }
}

File: backend/hrserver/src/main/java/com/beaconfire/hrserver/response/personalinfoResponses/EmergencyContact.java
package com.beaconfire.hrserver.response.personalinfoResponses;

import com.beaconfire.hrserver.domain.Address;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class EmergencyContact {
    //    private ResponseStatus status;
//    private Employee employee;
    private String fullName;
    private String phone;
    private Address address;
}
